shell/common/platform_view.cc

```diff
@@ -67,12 +67,26 @@ void PlatformView::NotifyDestroyed() {
   delegate_.OnPlatformViewDestroyed();
 }
 
-sk_sp<GrContext> PlatformView::CreateResourceContext() const {
+sk_sp<GrContext> PlatformView::CreateResourceContext() {
   FML_DLOG(WARNING) << "This platform does not setup the resource "
                        "context on the IO thread for async texture uploads.";
   return nullptr;
 }
 
+sk_sp<GrContext> PlatformView::GetOrCreateResourceContext() {
+  FML_DLOG(WARNING) << "This platform does not setup the resource "
+                       "context on the IO thread for async texture uploads.";
+  return nullptr;
+}
+
+fml::WeakPtr<GrContext> PlatformView::GetOrCreateWeakResourceContext() {
+  auto context = GetOrCreateResourceContext();
+  if (context) {
+    return fml::WeakPtrFactory<GrContext>(context.get()).GetWeakPtr();
```

Comment:
I don't think this is safe.  This WeakPtrFactory will be destructed immediately, resulting in invalidation of its weak pointers.

I think the PlatformView needs to hold a WeakPtrFactory allocated on the heap (similar to how `IOManager::resource_context_weak_factory_` worked)

Reply:
I think you're right, but I'm not sure we can safely create this on the PlatformView either.  It has to be created on the same thread as what will use it, and the methods for this class are documneted as being safe to access from any thread. I think I just have to move it back to IOManager - but it still seems fishy to me about how it's actually working there.

Reply:
Acutally I'm not sure that can make sense either.  I'm not seeing how we could use a WeakPtr for this at all - perhaps it would make more sense to just have consumers work with the `sk_sp` instead, and check if the `GrContext` is abandoned or null.